client/src/main/java/hudson/plugins/swarm/SwarmClient.java

```diff
@@ -261,6 +261,17 @@ protected void connect(Candidate target) throws InterruptedException {
         args.add("-headless");
         args.add("-noreconnect");
 
+        if(options.webSocket){
+            VersionNumber jenkinsVersion = new VersionNumber(launcher.VERSION);
```

Comment:
`launcher.VERSION` is the version of _Remoting_ (not Jenkins) on the Swarm client (i.e., [what is defined in the Swarm client build process in `pom.xml`](https://github.com/jenkinsci/swarm-plugin/blob/ab1858597444da458768848f174aac8c17be941d/client/pom.xml#L122-L126)). So this will typically be the latest version of Remoting (e.g., 4.3). It doesn't make sense to compare this to the Jenkins version since these are two different versioning schemes. And it doesn't make sense to examine the version of Remoting on the _client_ because the client will typically always be up-to-date. I'd suggest trying to find a way to compare this to the version of Remoting (or the Jenkins version) of the _server_. If that isn't practical, then a warning in the documentation would be acceptable. I'd also be curious as to the symptoms of the failure mode if WebSockets are used against a Jenkins master with a version of Remoting that doesn't support WebSockets (or whose support of WebSockets predates Remoting 4.3).

Reply:
Resolved in #222.